nuxeo-core/nuxeo-core-storage-mongodb/src/main/java/org/nuxeo/ecm/core/storage/mongodb/MongoDBChecker.java

```diff
@@ -18,27 +18,35 @@
  */
 package org.nuxeo.ecm.core.storage.mongodb;
 
+import java.io.File;
+import java.io.FileInputStream;
+import java.io.IOException;
+import java.io.InputStream;
 import java.util.concurrent.TimeUnit;
 
 import org.apache.commons.logging.Log;
 import org.apache.commons.logging.LogFactory;
 import org.bson.Document;
+import org.nuxeo.common.xmap.XMap;
 import org.nuxeo.launcher.config.ConfigurationException;
 import org.nuxeo.launcher.config.ConfigurationGenerator;
 import org.nuxeo.launcher.config.backingservices.BackingChecker;
+import org.nuxeo.runtime.model.Descriptor;
+import org.nuxeo.runtime.mongodb.MongoDBConnectionConfig;
+import org.nuxeo.runtime.mongodb.MongoDBConnectionHelper;
 
 import com.mongodb.MongoClient;
-import com.mongodb.MongoClientOptions;
-import com.mongodb.MongoClientURI;
 import com.mongodb.MongoTimeoutException;
-import com.mongodb.ServerAddress;
+import com.mongodb.client.MongoDatabase;
 
 public class MongoDBChecker implements BackingChecker {
 
     private static final Log log = LogFactory.getLog(MongoDBChecker.class);
 
     public static final String TEMPLATE_NAME = "mongodb";
 
+    public static final String CONFIG_NAME = "mongodb-connection-config.xml";
```

Comment:
This is really brittle as it depends on internal implementation details of the name of the config file in the mongodb template, which could change at any time. We really have to do better than these hacks in the future... Food for thought.

Reply:
Agreed that its not ideal, so I will `log.warn` if the config file isn't found.